### DISCUSSION TITLE
Why omitting pid in the perf_event_open API

### DISCUSSION THREAD

**Discussion by 0x161e-swei** [iovisor/bcc#4673] [Q&A]
I noticed that in [src/python/bcc/table.py](https://github.com/iovisor/bcc/blob/master/src/python/bcc/table.py#L1026), function `open_perf_event` does not expose the `pid` parameter in the API. Similarly in [src/cc/api/BPF.cc](https://github.com/iovisor/bcc/blob/master/src/cc/api/BPF.cc#L612), function `open_perf_event` does not expose the `pid` parameter in the API. However, in the [src/cc/libbpf.c](https://github.com/iovisor/bcc/blob/master/src/cc/libbpf.c) source code, `bpf_perf_open_event` offers the `pid` in its parameters.  
What was the reason behind not exposing the `pid` to the bcc pyhton or cc API?

**Comment by chenhengqi** [comment]
Not sure why it is omited in the first place. If you have such requirements, PR is welcome.

**Reply by 0x161e-swei** [reply]
Hi Hengqi, I opened a PR (https://github.com/iovisor/bcc/pull/4682) related to this discussion. Would you mind taking a look?
